app/src/main/java/com/nutomic/syncthingandroid/util/FileUtils.java

```diff
@@ -137,6 +136,19 @@ private static String getVolumePath(final String volumeId, Context context) {
         return null;
     }
 
+    @SuppressLint("ObsoleteSdkInt")
+    @TargetApi(21)
+    private static String volumeToPath(Object storageVolumeElement, Class<?> storageVolumeClazz) throws Exception {
+        try {
+            Method getDir = storageVolumeClazz.getMethod("getDirectory");
+            File file = (File) getDir.invoke(storageVolumeElement);
+            return file.getPath();
+        } catch (NoSuchMethodException e) {
+            Method getPath = storageVolumeClazz.getMethod("getPath");
+            return (String) getPath.invoke(storageVolumeElement);
+        }
+    }

```

Comment:
Would be useful to add a comment which one targets which API.

Reply:
Added some comments, not entirely sure on it. I just debugged a weird issue to finde `getPath` missing. `getDirectory` is documented as added in api level 30, but there's no information about `getPath`. I just kept that in case it's still the one present on some older android versions.